File: 020_Valid_Parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        
        e = ''
        list = []
        for c in s:
            if c=='(' or c=='[' or c=='{':
                list.append(c)
            elif c==')' or c==']' or c=='}':
                if not list:
                    return False
                e = list.pop()
                if e=='(' and c==')':
                    continue
                elif e=='[' and c==']':
                    continue
                elif e=='{' and c=='}':
                    continue
                else:
                    return False
        
        if not list:
            return True
        else:
            return False

File: 020_Valid_Parentheses.cpp
#include<stack>
class Solution {
public:
    bool isValid(string s) {
        
        std::stack<char> list;
        char c, e='0';
        
        for(int i=0; i<s.length(); i++){
            c = s[i];
            if(c=='('||c=='['||c=='{'){
                list.push(c);
            }
            else if(c==')'||c==']'||c=='}'){
                if(list.empty())
                    return false;
                e = list.top();
                list.pop();
                if(e=='('&&c==')')
                    continue;
                else if(e=='['&&c==']')
                    continue;
                else if(e=='{'&&c=='}')
                    continue;
                else
                    return false;
            }
        }
        
        if(list.empty())
            return true;
        else
            return false;
    }
};
